File: ios/YandexMobileAdsModule.m
#if !TARGET_OS_MACCATALYST

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_REMAP_MODULE(YandexMobileAds, YandexMobileAdsModule, NSObject)

RCT_EXTERN_METHOD(getLibraryVersion: (RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(initializeSdk: (RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(showDebugPanel: (RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(enableLogging: (BOOL)enableLogging)

RCT_EXTERN_METHOD(enableDebugErrorIndicator: (BOOL)enableIndicator)

RCT_EXTERN_METHOD(setLocationConsent: (BOOL)locationConsent)

RCT_EXTERN_METHOD(setAgeRestrictedUser: (BOOL)ageRestrictedUser)

RCT_EXTERN_METHOD(setUserConsent: (BOOL)consent)

@end

#endif

File: ios/YandexMobileAdsModule.swift
#if !targetEnvironment(macCatalyst)

import Foundation
import YandexMobileAds

@objc(YandexMobileAdsModule)
class YandexMobileAdsModule: NSObject {
    @objc
    func methodQueue() -> DispatchQueue {
        DispatchQueue.main
    }
    
    @objc
    func getLibraryVersion(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
        let libraryVersion = MobileAds.sdkVersion()
        resolve(libraryVersion)
    }
    
    @objc
    func initializeSdk(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
        MobileAds.initializeSDK {
            resolve(nil)
        }
    }
    
    @objc
    func showDebugPanel(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
        MobileAds.showDebugPanel()
        resolve(nil)
    }
    
    @objc
    func enableLogging(_ enableLogging: Bool) {
        if enableLogging {
            MobileAds.enableLogging()
        }
    }
    
    @objc
    func enableDebugErrorIndicator(_ enableIndicator: Bool) {
        /* Do something later */
    }
    
    @objc
    func setLocationConsent(_ locationConsent: Bool) {
        MobileAds.setLocationTrackingEnabled(locationConsent)
    }
    
    @objc
    func setAgeRestrictedUser(_ ageRestrictedUser: Bool) {
        MobileAds.setAgeRestrictedUser(ageRestrictedUser)
    }
    
    @objc
    func setUserConsent(_ consent: Bool) {
        MobileAds.setUserConsent(consent)
    }
}

#endif
